pages/glossary/semantic-role.md

```diff
@@ -7,13 +7,13 @@ input_aspects:
   - DOM tree
 ---
 
-A _semantic [role](https://www.w3.org/TR/wai-aria-1.1/#dfn-role)_ is a semantic association that indicates an object's type. This allows tools to present and support interaction with the object in a manner that is consistent with user expectations about other objects of that type.
+A _semantic association_, also called [role][], indicates an object's type. This allows tools to present and support interaction with the object in a manner that is consistent with user expectations about other objects of that type.
```

Comment:
This feels strange. The definition is of semantic role. However, the reader is presented with the definition of semantic association (which is also called semantic role). But 2 paragraphs below it is said that "The semantic role of an element is one way to give it a semantic association". Therefore, the semantic role is one way to give it the element a semantic role!!!

It seems to me that it would be much easier if we used just "semantic association", "explicit semantic role" and "implicit semantic role", and dropped the "semantic role" from our definitions. In the explicit and implicit definitions that's already the case, and they are much easier to read and understand than this one.

Reply:
* We need a definition of "semantic role". It is the result of a rather complex construction (conflict, explicit, implicit). We need a name for the result of this construction so we can reference it in rules.
* There are, indeed, two things named "role" (in ARIA). One is the concept (which I try here to rename as "semantic association", maybe not the best name). One is the construction that we make here. I tried to keep both inside the same definition as they are closely related, maybe that was a mistake and there should be two separate definitions instead.
* The logical dependencies are as follows:
  + A semantic association is something.
  + There exist different way of giving a semantic association to an element. Notably: explicit role, implicit role, and the complex construction called "semantic role".

  If the "semantic role" and "semantic association" are not somehow separated, then there is the (current) circularity of "a semantic role can be an explicit role which is a semantic role".


Would you say "semantic association" needs to be extracted into a separate definition? That is likely going to be a definition which is only used inside definitions… Not a huge problem, though…